openformats/formats/json.py

```diff
@@ -116,46 +102,51 @@ def _extract(self, parsed, nest=None):
         else:
             raise ParseError("Invalid JSON")
 
-    def _parse_special(self, key, value, value_position):
-        """
-        Parse a string that follows a subset of the the ICU message format
-        and return an OpenString object.
-
-        For the time being, only the plurals format is supported.
-        If `value` doesn't match the proper format, it will return None.
-        This method will also update the transcriber accordingly.
-
-        Note: if we want to support more ICU features in the future,
-        this would probably have to be refactored.
+    def _create_openstring(self, key, value, value_position):
+        """Return a new OpenString based on the given key and value
+        and update the transcriber accordingly based on the provided position.
 
         :param key: the string key
-        :param value: the serialized string that has all the content,
-            formatted like this (whitespace irrelevant):
-            { item_count, plural,
-                one { You have {file_count} file. }
-                other { You have {file_count} files. }
-            }
+        :param value: the translation string
         :return: an OpenString or None
         """
-        matches = re.match(
-            ur'\s*{\s*([A-Za-z-_\d]+)\s*,\s*([A-Za-z_]+)\s*,\s*(.*)}\s*', value
+        # First attempt to parse this as a special node, e.g. a pluralized string.
+        # If it cannot be parsed that way (returns None), parse it like
+        # a regular string.
+        icu_string = ICUParser.parse(key, value, value_position)
+        if icu_string:
```

Comment:
It's more Pythonic to use:
```suggestion
        if icu_string is not None:
```

Reply:
I'd argue that this is not the case. 

Moreover, the `parse()` method is expected to return an `ICUString` instance or `None`. It will never return any other object that may be evaluated to a boolean (e.g. an empty list), so I don't se a reason to use the more verbose version here, to be honest.